src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -123,34 +123,80 @@ SyntaxNodeAnalysisContext ctx
 				);
 
 				return;
-			}
-
-			// Literal arguments should always be named
-			foreach( var arg in unnamedArgs ) {
-
-				// Check if the argument type is literal
-				if( arg.Syntax.Expression is LiteralExpressionSyntax ) {
-
+			}
+
+			int numOfSwappableArgs = 0;
+			foreach( var arg in unnamedArgs ) {
+
+				// Literal expressions have their own issue
+				if( arg.Syntax.Expression is LiteralExpressionSyntax ) {
+					// Literal arguments should always be named
 					var fixerContext = CreateFixerContext( ImmutableArray.Create( arg ) );
-
 					ctx.ReportDiagnostic( Diagnostic.Create(
 							descriptor: Diagnostics.LiteralArgShouldBeNamed,
 							location: arg.Syntax.Expression.GetLocation(),
 							properties: fixerContext,
-							messageArgs: arg.ParamName
-						)
-					);
+							messageArgs: arg.ParamName ) );
+					continue;
+				}
+
+				// Take note of the type of argument
+				ITypeSymbol argType = GetArgumentType( ctx, arg.Syntax );
+
+				// Iterate through the other arguments in the list
+				foreach( var compareArg in unnamedArgs ) {
+					if( compareArg.Position == arg.Position ) {
+						continue;
+					}
+
+					// Get the associated parameter of the argument to compare
+					IParameterSymbol param = compareArg.Syntax.DetermineParameter(
+						ctx.SemanticModel,
+						allowParams: false );
+
+					// Check if the argument could apply to the other parameter
+					if( ctx.SemanticModel.Compilation.ClassifyConversion(
+						argType,
+						param.Type ).Exists ) {
+						// The argument can apply to multiple parameters,
+						// so take note of it
+						numOfSwappableArgs++;
+						break;
+					}
 				}
-			}
-
-			// TODO: if there are duplicate typed args then they should be named
-			// These will create a bit more cleanup. Fix should probably name
-			// all the args instead to avoid craziness with overloading.
+			}
+
+			// If more than one argument can apply to multiple parameters,
+			// then require the arguments to be named
+			if( numOfSwappableArgs > 1 ) {
+				var fixerContext = CreateFixerContext( unnamedArgs );
+
+				ctx.ReportDiagnostic( Diagnostic.Create(
+						descriptor: Diagnostics.NamedArgumentsRequired,
+						location: ctx.Node.GetLocation(),
+						properties: fixerContext ) );
+			}
+
+			// TODO: if there are duplicate typed args then they should be named
+			// These will create a bit more cleanup. Fix should probably name
+			// all the args instead to avoid craziness with overloading.

```

Comment:
I think you can remove this now...

Oh hey, overloading.... so my previous comment about only naming some of the args.... Hmmm....

Reply:
I don't think this concern is valid. Inserting argument names won't change which overload you use. It could be that we suggest that you explicitly pick an overload that you didn't want to, but already accidentally had (this would only be in VERY unlikely scenarios I think) **and** you still don't notice... and then the next person assumes you knew what you were doing... but that's kind of a general "problem" with this analyzer, so eh I don't buy it.

Sorry, just thinking out loud. I wrote this comment originally but that doesn't mean I remembered why immediately :)

Reply:
TL;DR: ignore the overload thing, let's do max one diagnostic per function call, and the fixerContext should have whatever args we deem worthy of naming, which may be an incomplete list.

Reply:
So is it to be removed, or not?